src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -1181,6 +1212,47 @@ private void addNewFolder() {
 		}
 	}
 
+	private void renameFolder() {
+		renameFolder.setEnabled(true);
+		IElement selected = list.getSelectedValue();
+		String newName = VOptionPane
+				.showInputDialog(mcreator, L10N.t("workspace.elements.folders.rename.message"), L10N.t("workspace.elements.folders.rename.title"), null, new OptionPaneValidatior() {
+					@Override public ValidationResult validate(JComponent component) {
+						String newName = ((JTextField) component).getText();

```

Comment:
How about checking for existing folders?

Reply:
done